code/modules/detectivework/samples.dm

```diff
@@ -163,17 +172,13 @@
 		S.forceMove(drop_location())
 	to_chat(user, span_notice("You transfer [length(S.evidence)] [length(S.evidence) > 1 ? "[evidence_type]s" : "[evidence_type]"] to \the [S]."))
 
-/obj/item/sample_kit/afterattack(atom/target, mob/user, proximity_flag, click_parameters)
-	. = ..()
-	if(!proximity_flag)
-		return
+/obj/item/sample_kit/interact_with_atom(atom/interacting_with, mob/living/user, list/modifiers)
```

Comment:
This item will also have issues with placing it on tables and in backpacks and the like, but since an intended behavior (seems to be) sampling those aforementioned items, you probably gotta figure out how you wanna address that on your own

Reply:
I've made left clicking default to going into storage/the table (via a horrible istable() check) and now you use right click to force a swab.